pilot/pkg/config/kube/gateway/controller.go

```diff
@@ -124,6 +143,33 @@ func (c controller) List(typ config.GroupVersionKind, namespace string) ([]confi
 	return nil, errUnsupportedOp
 }
 
+func (r *KubernetesResources) UpdateStatuses(c controller) {
+	c.handleStatusUpdates(r.GatewayClass)
+	c.handleStatusUpdates(r.Gateway)
+	c.handleStatusUpdates(r.HTTPRoute)
+	c.handleStatusUpdates(r.TCPRoute)
+	c.handleStatusUpdates(r.TLSRoute)
+	c.handleStatusUpdates(r.BackendPolicy)
+}
+
+func (c controller) handleStatusUpdates(configs []config.Config) {
+	for _, cfg := range configs {
+		ws := cfg.Status.(*kstatus.WrappedStatus)
+
+		if ws.Dirty {
+			_, err := c.cache.UpdateStatus(config.Config{
+				Meta:   cfg.Meta,
+				Status: ws.Unwrap(),
+			})
+			// TODO make this more resilient. When we add a queue we can make transient failures retry
+			// and drop permanent failures
```

Comment:
What happens during upgrade ( if ClusterRoles are not updated ? )

Reply:
In this case, since we drop updates, the status would not be updated and we would not retry. In the future we will likely make this more resilient and retry with queue, etc; when we do that we should detect permanent failures (like no RBAC) and drop them from retrying